src/main/java/com/mixpanel/android/util/HttpService.java

```diff
@@ -132,7 +132,7 @@ public boolean isOnline(Context context) {
                 }
                 retries = retries + 1;
             } catch (final IOException e) {
-                if (503 == connection.getResponseCode()) {
+                if (connection.getResponseCode() >= 500 && connection.getResponseCode() <= 599) {
```

Comment:
You should better use [constants](http://developer.android.com/reference/java/net/HttpURLConnection.html#HTTP_INTERNAL_ERROR) than magic numbers :)))